test/membrane_http_adaptive_stream/sink_test.exs

```diff
@@ -100,16 +108,88 @@ defmodule Membrane.HTTPAdaptiveStream.SinkTest do
     assert :ok = cleanup_fun.()
     assert_receive {SendStorage, :remove, %{type: :manifest, name: "index.m3u8"}}
     assert_receive {SendStorage, :remove, %{type: :manifest, name: "audio.m3u8"}}
-    assert_receive {SendStorage, :remove, %{type: :manifest, name: "video.m3u8"}}
+    assert_receive {SendStorage, :remove, %{type: :manifest, name: "video" <> _}}
     assert_receive {SendStorage, :remove, %{name: "audio_segment_1_" <> _}}
     assert_receive {SendStorage, :remove, %{name: "audio_segment_2_" <> _}}
     assert_receive {SendStorage, :remove, %{name: "video_segment_1_" <> _}}
     refute_receive {SendStorage, _, _}
+    Testing.Pipeline.stop_and_terminate(pipeline, blocking?: true)
+  end
+
+  test "audio and multiple video tracks" do
+    pipeline =
+      mk_pipeline([
+        {:video, "video_0"},
+        {:audio, "audio"},
+        {:video, "video_1"},
+        {:video, "video_2"}
+      ])
+
+    assert_receive {SendStorage, :store, %{type: :header, name: "audio_header" <> _}}
+    assert_receive {SendStorage, :store, %{type: :header, name: "video_header_" <> _}}
+    assert_receive {SendStorage, :store, %{type: :header, name: "video_header_" <> _}}
+    assert_receive {SendStorage, :store, %{type: :header, name: "video_header_" <> _}}
+    refute_receive {SendStorage, _, _}
+
+    send_buf(pipeline, "audio", 2)
+    assert_receive {SendStorage, :store, %{type: :manifest, name: "index.m3u8"}}
+    assert_receive {SendStorage, :store, %{type: :manifest, name: "audio.m3u8"}}
+    assert_receive {SendStorage, :store, %{name: "audio_segment_0_" <> _}}
+    refute_receive {SendStorage, _, _}
+    assert_pipeline_notified(pipeline, :sink, {:track_playable, "audio"})
+
+    send_buf(pipeline, "video_1", 3)
+    assert_receive {SendStorage, :store, %{type: :manifest, name: "index.m3u8"}}
+    assert_receive {SendStorage, :store, %{type: :manifest, name: "video_" <> _}}
```

Comment:
Looking at code here: https://github.com/membraneframework/membrane_http_adaptive_stream_plugin/blob/multiple_video_track_support/lib/membrane_http_adaptive_stream/hls.ex#L78
it seems that we know the exact name of the manifest file which is going to be updated, right?

Reply:
Same for similar cases below.

Reply:
Actually not. Message of type {SendStorage, _, _} dos not contain  of its source. Connection from track id to its source_id is not straightforward.